--- a/src/embedded_assets.rs
+++ b/src/embedded_assets.rs
@@ -3,7 +3,7 @@
 // SPDX-License-Identifier: MIT
 
 use base64::Engine;
-use proc_macro2::TokenStream;
+use proc_macro2::{Literal, TokenStream};
 use quote::{quote, ToTokens, TokenStreamExt};
 use sha2::{Digest, Sha256};
 use std::{
@@ -391,15 +391,19 @@
 impl ToTokens for EmbeddedAssets {
   fn to_tokens(&self, tokens: &mut TokenStream) {
     let mut assets = TokenStream::new();
-    for (key, (input, output)) in &self.assets {
+    for (key, (_input, output)) in &self.assets {
       let key: &str = key.as_ref();
-      let input = input.display().to_string();
-      let output = output.display().to_string();
-
-      // add original asset as a compiler dependency, rely on dead code elimination to clean it up
+      let bytes = std::fs::read(output).unwrap_or_else(|error| {
+        panic!(
+          "failed to read generated embedded asset {} because {}",
+          output.display(),
+          error
+        )
+      });
+      let bytes = Literal::byte_string(&bytes);
+
       assets.append_all(quote!(#key => {
-        const _: &[u8] = include_bytes!(#input);
-        include_bytes!(#output)
+        #bytes
       },));
     }
 
